Lib/test/test_peepholer.py

```diff
@@ -282,11 +283,17 @@ def test_binary_subscr_on_unicode(self):
         self.assertInBytecode(code, 'LOAD_CONST', 'f')
         self.assertNotInBytecode(code, 'BINARY_SUBSCR')
         self.check_lnotab(code)
+
         code = compile('"\u0061\uffff"[1]', '', 'single')
         self.assertInBytecode(code, 'LOAD_CONST', '\uffff')
         self.assertNotInBytecode(code,'BINARY_SUBSCR')
         self.check_lnotab(code)
 
+        code = compile('"\u0061\uffffxx"[1::2]', '', 'single')
+        self.assertInBytecode(code, 'LOAD_CONST', '\uffffx')
+        self.assertNotInBytecode(code,'BINARY_SUBSCR')
+        self.check_lnotab(code)
+
```

Comment:
Strictly speaking this test doesn't belong in `test_peepholer` now that the optimization is not part of the peephole optimiser.

Where are the other const folding tests?